src/Microsoft.Azure.WebJobs.ServiceBus/Bindings/ServiceBusBinding.cs

```diff
@@ -20,16 +20,19 @@ internal class ServiceBusBinding : IBinding
         private readonly IBindableServiceBusPath _path;
         private readonly IAsyncObjectToTypeConverter<ServiceBusEntity> _converter;
         private readonly AccessRights _accessRights;
+        private readonly bool _isQueue;
 
-        public ServiceBusBinding(string parameterName, IArgumentBinding<ServiceBusEntity> argumentBinding, ServiceBusAccount account, IBindableServiceBusPath path, AccessRights accessRights)
+        public ServiceBusBinding(string parameterName, IArgumentBinding<ServiceBusEntity> argumentBinding, ServiceBusAccount account, IBindableServiceBusPath path, ServiceBusAttribute attr)
         {
             _parameterName = parameterName;
             _argumentBinding = argumentBinding;
             _account = account;
             _namespaceName = ServiceBusClient.GetNamespaceName(account);
             _path = path;
-            _accessRights = accessRights;
-            _converter = CreateConverter(account, path, accessRights);
+            _accessRights = attr.Access;
+            _isQueue = !(attr is ServiceBusTopicAttribute);
```

Comment:
>!(attr is ServiceBusTopicAttribute); [](start = 23, length = 36)

nit: type checks like this can get brittle - somebody else adds another derived class; or the type hierarchy gets juggled - it can easily break. 
Consider an internal virtual IsQueue property on ServiceBusAttribute.